### PR TITLE
Implement api version updates via yaml parsing

### PR DESCRIPTION
**Pull request by tstirrat15**
## Description
I'd rather use `buf generate` directly and use a normal `buf.gen.yaml` file than have a hashbang at the top of the file. This implements that by doing yaml parsing and modifying the associated node.

I also changed how this action is built and packaged, so it no longer needs to commit its `node_modules`.

## Changes
* Add new variant of buf api update action
* Use vitest instead of jest
* Use ncc for bundling: https://github.com/vercel/ncc
* Get rid of committed `node_modules`
* Simplify tsconfig
* Update node version used by action
* Add `node_modules` to gitignore

## Testing
I'll execute the action from another lib and validate that it works.

### CODE REVIEW

**Comment by tstirrat15** [comment]
This is evidence that it's working: https://github.com/authzed/authzed-rb/actions/runs/11636989379/job/32409402272?pr=136

I'm gonna finish up this PR with the other package that's based on node.

File: buf-api-update/dist/index.js
Comment (tstirrat15) [inline buf-api-update/dist/index.js:1]: Autogen file

File: buf-api-update/lib/handlers.js
Comment (tstirrat15) [inline buf-api-update/lib/handlers.js:1]: Replaced by dist file

File: buf-api-update/jest.config.js
Comment (tstirrat15) [inline buf-api-update/jest.config.js:1]: Replaced by vitest use

File: buf-api-update/src/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as core from '@actions/core';
-import fs from 'fs';
-import util from 'util'
+import * as fs from 'fs';
+import * as util from 'util'
```
Comment (tstirrat15) [inline buf-api-update/src/index.ts:3]: This keeps TS happy

File: buf-api-update/src/handlers.ts [hunk 2]
```diff
@@ -22,4 +24,18 @@ export const fileFormatHandlers: Record<FileFormatType, Handler> = {
...
+              if (pair.key.value === "module") {
+                  if (pair.value) {
+                      pair.value.value = pair.value?.value.replace(/:.*$/, `:${apiCommit}`)
+                  }
+              }
+          }) as visitorFn<Pair>)
+      })
+      const updated = doc.toString()
+      return [updated, contents !== updated]
+  }
```
Comment (tstirrat15) [inline buf-api-update/src/handlers.ts:40]: This is the meat of the change for this PR

File: buf-api-update/tsconfig.json [hunk 3]
```diff
@@ -1,12 +1,9 @@
 {
     "compilerOptions": {
-      "target": "es6",
-      "module": "commonjs",
-      "outDir": "./lib",
-      "rootDir": "./src",
+      "lib": ["ES2023"],
+      "module": "node16",
+      "target": "ES2022",
       "strict": true,
```
Comment (tstirrat15) [inline buf-api-update/tsconfig.json:6]: From https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-20.

Keeping the file as minimal as possible should mean we break things as seldom as possible.

**Review by tstirrat15 — COMMENTED** [review]
See comments

File: buf-api-update/src/handlers.ts [hunk 4]
```diff
@@ -22,4 +24,18 @@ export const fileFormatHandlers: Record<FileFormatType, Handler> = {
     const updatedContents = contents.replace(regex, `def authzedProtoCommit = "${apiCommit}"`);
     return [updatedContents, contents !== updatedContents]
   },
+  'buf-gen-yaml': (contents: string, apiCommit: string) => {
+      const doc = parseDocument(contents)
```
Comment (josephschorr) [inline buf-api-update/src/handlers.ts:28]: Make sure this is using the same parsing; some YAML libs have unsafe versions

Comment (josephschorr) [reply buf-api-update/src/handlers.ts:28]: Also add error handling here

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**